core/src/backend/audio/decoders/adpcm.rs

```diff
@@ -95,108 +39,81 @@ impl<R: Read> AdpcmDecoder<R> {
         }) as usize
             + 2;
 
+        let num_channels = if is_stereo { 2 } else { 1 };
+
         Self {
             inner: reader,
             sample_rate,
-            is_stereo,
             bits_per_sample,
             sample_num: 0,
-            left_sample: 0,
-            left_step: 0,
-            left_step_index: 0,
-            right_sample: 0,
-            right_step: 0,
-            right_step_index: 0,
-            decoder: Self::SAMPLE_DELTA_CALCULATOR[bits_per_sample - 2],
+            channels: (0..num_channels)
```

Comment:
Could derive `Clone` and `Defualt` for `Channel` and then `vec![Default::default(); num_channels]`